modules/Movecraft/src/main/java/net/countercraft/movecraft/sign/RepairSign.java

```diff
@@ -1,6 +1,265 @@
 package net.countercraft.movecraft.sign;
 
+import com.sk89q.worldedit.Vector;
+import com.sk89q.worldedit.blocks.BaseBlock;
+import com.sk89q.worldedit.extent.clipboard.Clipboard;
+import net.countercraft.movecraft.Movecraft;
+import net.countercraft.movecraft.MovecraftLocation;
+import net.countercraft.movecraft.MovecraftRepair;
+import net.countercraft.movecraft.config.Settings;
+import net.countercraft.movecraft.craft.Craft;
+import net.countercraft.movecraft.craft.CraftManager;
+import net.countercraft.movecraft.localisation.I18nSupport;
+import net.countercraft.movecraft.mapUpdater.update.UpdateCommand;
+import net.countercraft.movecraft.mapUpdater.update.WorldEditUpdateCommand;
+import net.countercraft.movecraft.repair.Repair;
+import net.countercraft.movecraft.repair.RepairManager;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.Sign;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.block.SignChangeEvent;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.InventoryHolder;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.*;
 
 public class RepairSign implements Listener{
+    private String HEADER = "Repair:";
+    private HashMap<UUID, Long> playerInteractTimeMap = new HashMap<>();//Players must be assigned by the UUID, or NullPointerExceptions are thrown
+
+    @EventHandler
+    public void onSignChange(SignChangeEvent event){
+        if (!event.getLine(0).equalsIgnoreCase(HEADER)){
+            return;
+        }
+        if (event.getLine(1).isEmpty()){
+            event.getPlayer().sendMessage("You must specify a repair state name on second line");
+            event.setCancelled(true);
+            return;
+        }
+    }
+
+    @EventHandler
+    public void onSignClick(PlayerInteractEvent event){
+        String firstLine = "";
+        if (event.getAction() == Action.RIGHT_CLICK_BLOCK || event.getAction() == Action.LEFT_CLICK_BLOCK) {
```

Comment:
(opinion) It may make the most sense to return here if the action is not a right or left click, then go through the logic for each action.

Ex:
if(!right || !left)
  return

if right
  *starting repair*
else
  *saving state*